### ISSUE TITLE
GUI : System Logs

### ISSUE THREAD

**Issue by derick-montague** [ibm-openbmc/dev#2626]
## SME
**BMC**: @santoshpuranik 
**RAS**: George Ahrens
**PHYP**:

## Key conversations
- Event logs schema proposal v2: https://ibm.ent.box.com/notes/791481299908?s=0xba8qx3f8tjyijulfp2c6n8kb70snyb
- As per Matt Spinler, the above schema is final except for Service Action Flag (and whether that will be integrated into Severity field or will be an OEM field). [Chat link.](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1618512784234900?thread_ts=1612269693.279100&cid=C0Q6TQP5Z)
- Progress logs API chat: [Chat link ](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1618841538274900?thread_ts=1612354564.303900&cid=C0Q6TQP5Z)

### Summary
There are multiple log types and it is not clear if they will be able to live within the same table or even within the same page. Redfish has a logs service and each log type listed is a different service call. We will need to determine if the data in the response will work in the same table structure. Another potential issue is performance, having all of these logs within one table will require calling, consolidating, and sorting all the responses from the different log service API calls.

- Audit Logs: /redfish/v1/Managers/BMC/LogServices/Audit/ (low priority due to timeline)
- Maintenance: /redfish/v1/Managers/BMC/LogServices/Maintenance
- Progress/Boot Logs: /redfish/v1/Managers/BMC/LogServices/Progress (related #1303, #2843)
- Hidden: /redfish/v1/Managers/BMC/LogServices/Hidden 
- Event: /redfish/v1/Managers/BMC/LogServices/Event

_Refer to the summary section System Logs Epic for detail about each log type #1523_

### User Story
As a <type of user>
I need to <perform some activity>
in order to <complete some task or accomplish some goal>

### InVision Prototype
- [ ] Event logs: Check https://github.com/ibm-openbmc/dev/issues/2156
- [ ] Progress logs:  Check https://github.com/ibm-openbmc/dev/issues/3070

### Tasks
- [ ] Create design review story in webui-vue repo
- [ ] Send email to community sharing the design review story
- [ ] Post in story in the discord webui channel

## Notes
**Audit Logs**
> I'd recommend separating out things like the machine power state changing, from the user triggering it, so Support can find out when a box crashed/ went down/ came up, in relation to the problem.
My plan to deal with Audit logs was to have them in their own LogService, so when we collect service data we don't have to scrub the logs for GDPR compliance, just not collect the audit log.
~ Justin Thaler

**Maintenance Logs**
> This is another way to get to the following screens: 
> Deconfiguration (Guard) Records and Hardware Serviceable Events screens
~ Nicole Conser

**Previous Boot Progress Indicator & Progress Indicator History**
> Even though these are separate menu items in ASMI, they are covered by one Redfish LogService. 
    ~Gunnar:  [7/15 Slack convo](https://ibm-systems-power.slack.com/archives/GBVDZRCQK/p1594764817141200)

**Event/Error logs**
See the discovery box notes for a link to a recorded call on 1/12/21 with the SEET group and link to slack conversation in design channel.

## References
- [Slack conversation](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1592239346365500?thread_ts=1592238708.364400&cid=C0Q6TQP5Z)
    > We were talking about if all the logServices (Audit, Maintenance, Progress (aka Boot), Correctable (aka Hidden)) should be on one page on the GUI and how should that be displayed ? Do you have any thoughts ? FYI
- [Redfish log examples](https://ibm.box.com/s/gtk6w0psfvqletup8bt4ie2fh1ihjwbu)
- [Discovery folder](https://ibm.box.com/s/qd4iw5xsfpaeeyvm4a8k4j5drnd9h8r4)
- [Discovery box note](https://ibm.box.com/s/dd1d7lq5pcfzxboy1abxp74o3s9qe7u7)

# FED
### UX Flow / Interaction Requirements
- [ ] Feature breakdown

### cURL Commands and Redfish Response
- [ ] Add Redfish response
- [ ] Add cURL commands

### Notes
- [ ] Any known API workarounds
- [ ] Any server-side functionality WIP

#### Browser Tests (Chrome, Firefox, Safari (Mac), Edge (Windows))
- [ ] Tested responsive layout on supported viewport sizes
- [ ] Verified functionality works as expected, e.g. success and failure use cases
- [ ] Resolved any JavaScript errors thrown to the console

#### Accessibility Tests
- [ ] Verified markup is semantic
- [ ] No reported issues in IBM Accessibility Checker (Free open source tool) https://www.ibm.com/able/toolkit/tools/

#### Test Hooks
- [ ] Verify need for `data-test-id` hooks with test team

**Comment by derick-montague** [comment]
Audit logs may be pushed out.

**Comment by nicoleconser** [comment]
Matt Spinler gave an update on **Event Log Display** in a recent SEET meeting…  I listened to the first ~40 mins and took some notes — see notes + recording URL in [Slack](https://ibm-systems-power.slack.com/archives/GBVDZRCQK/p1610491473128400). @yoshiemuranaka @derick-montague

**Comment by derick-montague** [comment]
> @spinler gave an update on **Event Log Display** in a recent SEET meeting… I listened to the first ~40 mins and took some notes — see notes + recording URL in [Slack](https://ibm-systems-power.slack.com/archives/GBVDZRCQK/p1610491473128400). @yoshiemuranaka @derick-montague

Thank you @nicoleconser. I also added a Systems Logs discovery doc and folder and linked in the description. I have included the recording info there also since only the design team has access to that slack conversation.

**Comment by derick-montague** [comment]
[Slack conversation about logs timeline](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1612269693279100)

**Comment by ParishrutB** [comment]
[An ongoing discussion about how progress codes implementation](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1614011304018400)...

Summary:
- Reference code (e.g. C700406E) currently does not come directly from the api call. The GUI is required to parse it and extract this code.
- Discussion regarding an alternate approach to fix this is still ongoing in the backed.

**Comment by derick-montague** [comment]
I don't believe that this is correct `Progress/Boot Logs: /redfish/v1/Managers/BMC/LogServices/Progress`. I believe it is `Progress/Boot Logs: /redfish/v1/Managers/BMC/LogServices/PostCodes/Entries (related #1303, #2843)`. There will be a property that the use can download the full file.
```
{
  "@odata.id": "/redfish/v1/Systems/system/LogServices/PostCodes/Entries",
  "@odata.type": "#LogEntryCollection.LogEntryCollection",
  "Description": "Collection of POST Code Log Entries",
  "Members": [
    {
      "@odata.id": "/redfish/v1/Systems/system/LogServices/PostCodes/Entries/B1-1",
      "@odata.type": "#LogEntry.v1_4_0.LogEntry",
      "Created": "2021-02-09T01:38:32+00:00",
      "EntryType": "Event",
      "Id": "B1-1",
      "Message": "Boot Count: 1: TS Offset: 0.0000; POST Code: 0x4331303031463030",
      "MessageArgs": [
        "1", <--- Boot Count
        "0.0000", <--- Time Stamp Offset
        "0x4331303031463030" <--- Progress Code
      ],
      "MessageId": "OpenBMC.0.1.BIOSPOSTCode",
      "Name": "POST Code Log Entry",
      "Severity": "OK",
      "AdditionalDataURI" : "/redfish/v1/Systems/system/LogServices/PostCodes/attachment/B1-1"
    }
```

Also, the PEL will be added to the `AdditionalDataURI` property and we need to be able to download that from the GUI.

**Comment by ParishrutB** [comment]
Waiting to get some data from backend so that we can mock up the GUI - [Reference chat](https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1614218702165200?thread_ts=1614099917.073200&cid=C0Q6TQP5Z)

**Comment by derick-montague** [comment]
We are blocked waiting on backend to determine what will make it into the Redfish spec. Pari and Priyanka are going to create low fidelity wire frames to help the engineering team understand user needs the design team discovered.

**Comment by derick-montague** [comment]
@ParishrutB and @priyanka-pillai97 discussed a concern today with decisions being discussed. The BMC SME is @spinler 

There is discussion that if there is a HW issue that triggers an event, that event will include only the location code and not include the Part # or Serial #.  It has been suggested that the user will have to look at the VPD to determine this information. This puts the following user need at risk.

```
As a user that is verifying that a FRU has been replaced,
I need the serial number, part number, and location code in the event log created by the faulty FRU
So that I can compare the serial and/or part number for the FRU in the location sent in the event log
```
If this is an issue of scope, we understand that constraint and would like to acknowledge this will change the users workflow.

If the user is not able to get the part number and/or serial number from the event log, they will have to:

1. Get the location code from the event log
2. Go to the HW Inventory and Status page and find the FRU at that location code
3. Write down the Part and/or Serial numbers
4. Use the documented VPD to later compare and verify the successful replacement of the FRU

The users included in research verified the importance of including the VPD info in the Event log for competing this use case efficiently and effectively.

**Comment by ParishrutB** [comment]
Event logs - as per Matt Spinler - RAS team still needs to get back regarding how Service Action Flag will be implemented - whether it will be an OEM field or will be integrated with Severity field.
